src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -322,5 +322,51 @@ internal static IOperation GetRootOperation(this IOperation operation)
 
             return operation;
         }
+
+        /// <summary>
+        /// Gets either a loop or a switch operation that corresponds to the given branch operation.
+        /// </summary>
+        /// <param name="operation">The branch operation for which a corresponding operation is looked up</param>
+        /// <returns>The corresponding operation or <c>null</c> in case not found (e.g. no loop or switch syntax)</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="operation"/> is null</exception>
+        /// <exception cref="InvalidOperationException">The operation is a part of Control Flow Graph or it has an invalid branch kind. 
+        /// Applicable kinds: <see cref="BranchKind.Break"/> and <see cref="BranchKind.Continue"/>.</exception>
+        public static IOperation GetCorrespondingOperation(this IBranchOperation operation)
+        {
+            if (operation == null)
+            {
+                throw new ArgumentNullException(nameof(operation));
+            }
+
+            if (operation.SemanticModel == null)
+            {
+                throw new InvalidOperationException(CodeAnalysisResources.OperationMustNotBeControlFlowGraphPart);
+            }
+            
+            if (operation.BranchKind != BranchKind.Break && operation.BranchKind != BranchKind.Continue)
+            {
+                throw new InvalidOperationException(
+                    string.Format(CodeAnalysisResources.InvalidBranchKindForFindingCorrespondingOperation, operation.BranchKind));
+            }
+
+            if (operation.Target == null)
+            {
+                return null;
+            }
+            
+            for (IOperation current = operation; current.Parent != null; current = current.Parent)
+            {
+                switch (current)
+                {
+                    case ILoopOperation correspondingLoop when operation.Target.Equals(correspondingLoop.ExitLabel)
+                                                               || operation.Target.Equals(correspondingLoop.ContinueLabel):
```

Comment:
Nit: Would prefer the `||` on the other line, so the operation.Target.Equals lines up.